server/PathServices/SignalRRealtimeDataRepository.cs

```diff
@@ -41,12 +43,36 @@ public SignalRRealtimeDataRepository(IPathDataRepository pathDataRepository, IPa
         /// <returns>A collection of arriving trains.</returns>
         public Task<IEnumerable<RealtimeData>> GetRealtimeData(Station station)
         {
-            return Task.FromResult(this.GetRealtimeData(station, RouteDirection.ToNY).Union(this.GetRealtimeData(station, RouteDirection.ToNJ)).Where(data => data.DataExpiration > DateTime.UtcNow));
+            var allData = this.GetRealtimeData(station, RouteDirection.ToNY).Union(this.GetRealtimeData(station, RouteDirection.ToNJ));
+            var freshData = allData.Where(dataPoint => dataPoint.DataExpiration > DateTime.UtcNow);
```

Comment:
Interesting. Worried that this condition will always be met when the headways are really large... it may be fine. Need to double check.